### DISCUSSION TITLE
Elsa 3 distributed activity runners?

### DISCUSSION THREAD

**Discussion by rboylesDev** [elsa-workflows/elsa-core#4757] [Q&A]
I've been playing with Elsa 3 and I cannot get any activities running on multiple Elsa nodes. I have Elsa 3 server running and a console app wired up the same way. I'm using the Mass Transit Rabbit MQ dispatcher, but my main server node is the only one that ever picks up workflows and activities. What I'm really looking for is a guide on how to actually have this dispatch work to multiple nodes to execute the workflows, as that will be critical for scaling the system.

**Comment by sfmskywalker** [comment]
The distribution of work that consumes the queue from RabbitMQ is handled by RabbitMQ, so I don't think there's anything else you would have to do since Elsa relies on MassTransit and its consumer model. That being said, documentation for these kinds of setups will come, hopefully sooner rather than later.

**Comment by rboylesDev** [comment]
One thing I did find that made it work. I **have** to include the Use Workflow Management to be able to run any workflow on the worker. If I don't specify this, even though I have the workflow runtime and dispatcher, the worker tries to load everything from memory. Obviously, the workflows do not exist there.

``` c#
elsa.UseWorkflowManagement(management => management.UseEntityFrameworkCore(e => e.UseSqlServer(sqlConnection)));
```

I have also found that my worker will fail running a workflow if it does not have all the activities that the workflow contains. This meant I need Timer activities registered on workers to be able to run workflows that are triggered by timer, even if a different node triggered it. 

Timers then became a problem because if I just call UseScheduling, all nodes will trigger the timer. I had to explicitly call AddActivitiesFrom<Elsa.Scheduling.Features.SchedulingFeature>(); to get it to reference the activity without wiring in the triggering.
